### ISSUE TITLE
The ISO8601 Standard is being updated

### ISSUE THREAD

**Issue by maggiepint** [moment/moment#3925]
[The ISO8601 standard is being updated](https://www.iso.org/news/2017/02/Ref2164.html)

What do we need to do to be prepared for these changes? As a representative of TC39, I may get in contact with them to see if they can give me more help on the date standard side. From moment's side, we probably just have to prepare to parse more things.

Step 1: see if we can get a copy of the upcoming standard from somewhere, and an idea of the release timeline.

**Comment by icambron** [comment]
I think we should do nothing at all until it's published. It's gonna be a while before this is a thing. Standards move slowly and so does adoption.

We're not even fully compliant with the existing spec (or if I'm honest, with my reading of the Wikipedia article about the spec). There are all sorts of things not supported by Moment. We don't worry about them because either no one uses them or they're not compatible with JS dates (i.e. an hour/minute/second without a date part). My point isn't that we shouldn't be willing to parse more strings, but that I'm not sweating compliance with a spec as end in itself.

Some of this does sound like it fits in just fine:

> there are a lot of common expressions that don’t have a shared definition: semester, trimester, spring, summer, autumn, winter, to name a few. For these everyday terms, and questions such as ‘when does a day start, or end’, we’ve developed new ways of representation."”

So if that's like (I'm literally making up the spec on the spot here): `1999-11-01SOD`, then I agree that's just some strings we gotta parse and maybe some support for things like `startOf('isoworkday')` and `endOf('trimester')`. I like that stuff for the same reason I like ISO weeks.

OTOH, this sounds like it's not in Moment's scope:

> Klaus goes on to explain that DIS 8601-2, extensions, also covers: how dates are grouped; how date ranges are specified; how intervals between recurring events need to be represented, and how to deal correctly with leap-seconds.

Ranges, groups, and recurrence all sound like they needn't be in Moment. Leap seconds sounds complicated and relevant but not possible platformwise. I don't think we should worry about whether Moment's model is expansive enough to capture all of the concepts in the new spec. I think we'll just add some feature where they make sense.

TLDR: shrug. Let's wait and see.

**Comment by marwahaha** [comment]
See https://momentjs.com/docs/#/-project-status/
